js/components/Sources.js

```diff
@@ -1,27 +1,65 @@
 "use strict";
 
 const React = require("react");
+const {DOM: dom} = React;
 const { bindActionCreators } = require("redux");
 const { connect } = require("react-redux");
 const actions = require("../actions");
-const dom = React.DOM;
+const { getSelectedSource } = require("../queries");
+const { groupBy, values } = require("../utils");
 
 require("./Sources.css");
 
-function Sources({ sources, selectSource }) {
-  const sourceArr = Object.keys(sources).map(k => sources[k]);
+/**
+ * Takes a sources object indexed by actor and
+ * returns a sources object indexed by source domain.
+ *
+ * @returns Object
+ */
+function groupSourcesByDomain(sources) {
+  return groupBy(values(sources), source => {
+    if (!source.url) {
+      return source.introductionType;
```

Comment:
`introductionType` comes from the JS engine and tell you what kind of source it is (eval, new Function, script tag, etc). Many scripts don't have a URL if they are eval'ed. You see this on any complex site; the list of sources includes things like `Function`, `eval`, etc. Eventually we will want to treat this differently.